lib/multi.c

```diff
@@ -1079,6 +1087,12 @@ CURLMcode curl_multi_wait(struct Curl_multi *multi,
       }
     }
   }
+  else if(toofast_flag == 1 && timeout_internal > 0) {
+    /* If all socket descriptors are in a 'CURLM_STATE_TOOFAST' state,
+       we should wait for an exact timeout. Otherwise it will
+       be mistaken for busy-looping*/
+    Curl_wait_ms((int)timeout_internal);
```

Comment:
No, libcurl should *never* wait in a blocking manner. The waiting should be done by the application as the socket should be removed from the select() call until it is time to transfer again. Applications should not busy-loop because of that. That's most likely a bug in the application or possibly in libcurl.